generatebundlefile/main.go

```diff
@@ -216,13 +219,43 @@ func main() {
 			if err != nil || dockerAuth.Authfile == "" {
 				BundleLog.Error(err, "Unable create AuthFile")
 			}
-			err = dockerAuth.Remove()
 			if err != nil {
 				BundleLog.Error(err, "Unable remove AuthFile")
 			}
 			for _, charts := range addOnBundleSpec.Packages {
 				err = clients.PromoteHelmChart(charts.Source.Repository, dockerAuth.Authfile, true)
 			}
+			err = dockerAuth.Remove()
+			return
+		}
+
+		// See above comment about compound check when we want to cutover
+		// if o.publicProfile != "" && if o.privateProfile != "" {}
+		if o.privateProfile != "" {
+			BundleLog.Info("Starting release to private ECR process....")
+			clients, err := GetSDKClients()
+			clients, err = clients.GetProfileSDKConnection("ecr", o.privateProfile)
+			clients, err = clients.GetProfileSDKConnection("sts", o.privateProfile)
+			if err != nil {
+				BundleLog.Error(err, "Unable create SDK Client connections")
+			}
+			dockerReleaseStruct = &DockerAuth{
+				Auths: map[string]DockerAuthRegistry{
+					fmt.Sprintf("%s.dkr.ecr.%s.amazonaws.com", clients.stsClient.AccountID, ecrRegion):        DockerAuthRegistry{clients.ecrClient.AuthConfig},
+					fmt.Sprintf("%s.dkr.ecr.%s.amazonaws.com", clients.stsClientRelease.AccountID, ecrRegion): DockerAuthRegistry{clients.ecrClientRelease.AuthConfig},
+				},
+			}
+			dockerAuth, err = NewAuthFile(dockerReleaseStruct)
+			if err != nil || dockerAuth.Authfile == "" {
```

Comment:
```suggestion
			if err != nil {
```

Seems like that case should just generate an err response from NewAuthFile